### ISSUE TITLE
perl no longer installs un-versioned binaries

### ISSUE THREAD

**Issue by vsoch** [LLNL/spack#29409]
For singularity HPC, we create recipes that can confidently say the path and name of a binary in advance. E.g., for our perl recipe we have:

```yaml
aliases:
  perl: /opt/view/bin/perl
  perlbug: /opt/view/bin/perlbug
  perldoc: /opt/view/bin/perldoc
  perlivp: /opt/view/bin/perlivp
  perlthanks: /opt/view/bin/perlthanks
```
As of perl 5.34, this tree (and the subsequent view) turns out correctly - e.g., I have a perl / perlthanks and all the expected executables in bin.
```bash
$ ls opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/perl-5.34.0-bvgnm2ejnajpvaruta22d5c24g6qi4zu/bin/
corelist  h2ph       perl        perlthanks  pod2text    ptardiff   xsubpp
cpan      h2xs       perl5.34.0  piconv      pod2usage   ptargrep   zipdetails
cpanm     instmodsh  perlbug     pl2pm       podchecker  shasum
enc2xs    json_pp    perldoc     pod2html    prove       splain
encguess  libnetcfg  perlivp     pod2man     ptar        streamzip
```
However with 5.35.0 I no longer have any un-versioned binaries. They all end in the version.

```bash
 ls opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/perl-5.35.0-smm2auqgf4jwkvi7uzvrvjmqdxcwqtkc/bin/
corelist5.35.0   json_pp5.35.0     pl2pm5.35.0       ptardiff5.35.0
cpan5.35.0       libnetcfg5.35.0   pod2html5.35.0    ptargrep5.35.0
cpanm            perl5.35.0        pod2man5.35.0     shasum5.35.0
enc2xs5.35.0     perlbug5.35.0     pod2text5.35.0    splain5.35.0
encguess5.35.0   perldoc5.35.0     pod2usage5.35.0   streamzip5.35.0
h2ph5.35.0       perlivp5.35.0     podchecker5.35.0  xsubpp5.35.0
h2xs5.35.0       perlthanks5.35.0  prove5.35.0       zipdetails5.35.0
instmodsh5.35.0  piconv5.35.0      ptar5.35.0
```
I suspect this was implemented to make them install nicely alongside one another, but I consider it a bug that the install cannot output the plain binary. If this was done at the level of perl (e.g., outside of spack) I'd want to suggest that (for spack) we go ahead and create links so that the correct names are present. I think this should be the default (what is expected) and turned off with a variant if someone has issue. Thanks!

**Comment by vsoch** [comment]
Please see https://github.com/spack/spack/pull/29413 for my suggested fix.

**Comment by alalazo** [comment]
Fixed by #29456
